tests/test_ext_marshmallow.py

```diff
@@ -33,6 +35,42 @@ def test_can_use_schema_as_as_definition(self, spec, schema):
         assert props['id']['type'] == 'integer'
         assert props['name']['type'] == 'string'
 
+
+class TestCustomField:
+
+    def test_can_use_custom_field_decorator(self, spec):
+
+        @swagger.map_to_swagger_field(DateTime)
+        class CustomNameA(Field):
+            pass
+
+        @swagger.map_to_swagger_field('integer', 'int32')
+        class CustomNameB(Field):
+            pass
+
+        class CustomPetASchema(PetSchema):
+            name = CustomNameA()
+
+        class CustomPetBSchema(PetSchema):
+            name = CustomNameB()
+
+        spec.definition('Pet', schema=PetSchema)
+        spec.definition('CustomPetA', schema=CustomPetASchema)
+        spec.definition('CustomPetB', schema=CustomPetBSchema)
+
+        props_0 = spec._definitions['Pet']['properties']
+        props_a = spec._definitions['CustomPetA']['properties']
+        props_b = spec._definitions['CustomPetB']['properties']
+
+        assert props_0['name']['type'] == 'string'
+        assert 'format' not in props_0['name']
+
+        assert props_a['name']['type'] == 'string'
+        assert props_a['name']['format'] == 'date-time'
+
+        assert props_b['name']['type'] == 'integer'
+        assert props_b['name']['format'] == 'int32'
+
```

Comment:
I was concerned because you originally raised an `ArgumentError` and I didn't think `ArgumentError` was a built-in. In any case, I'd like to have that tested in order to define the expected behavior before this is merged.